bowtie/_cli.py

```diff
@@ -1684,6 +1708,623 @@ async def info(
+            implementation = report.implementations[id]
+            version = implementation.version or "latest"
+            unsuccessful = report.unsuccessful(id)
+            inner_table_rows.append(
+                create_inner_table_row(version, unsuccessful),
+            )
+
+        inner_markdown_table = convert_table_to_markdown(
+            inner_table_columns,
+            inner_table_rows,
+        )
+        row_data = [
+            dialect.pretty_name,
+            inner_markdown_table,
+            f"**{report.total_tests} {test} ran**",
+        ]
+        rows_data.append(row_data)
+
+    final_content: Iterable[str] = []
+    for _, row_data in enumerate(rows_data):
+        final_content.append(
+            f"### Dialect: {row_data[0]}\n{row_data[1]}\n\n{row_data[2]}",
+        )
+
+    return f"## Trend Data of {id} versions:\n\n" + "\n\n".join(final_content)
+
+
+class _VersionedReportsTar(click.File):
+    """
+    Select a tar containing previously produced versioned Bowtie reports.
+    """
+
+    name = "versioned_reports_tar"
+    mode = "rb"
+
+    def convert(  # type: ignore[reportIncompatibleMethodOverride]
+        self,
+        value: str | PathLike[str] | IO[Any],
+        param: click.Parameter | None,
+        ctx: click.Context,
+    ) -> tuple[frozenset[str], Iterable[tuple[str, Dialect, _report.Report]]]:
+        input = super().convert(value, param, ctx)
+
+        connectable = cast(
+            _connectables.Connectable,
+            ctx.params.get("connectable"),
+        )
+        id = connectable.to_terse()
+
+        dialects = cast(Iterable[Dialect] | None, ctx.params.get("dialects"))
+
+        try:
+            with tarfile.open(fileobj=input) as tar:
+                members = tar.getmembers()
+
+                if not any(
+                    member.name.startswith(f"./{id}/") for member in members
+                ):
+                    STDERR.print(
+                        f"Couldn't find a '{id}' directory in {input.name}.",
+                    )
+                    ctx.exit(EX.DATAERR)
+
+                versions: frozenset[str] = frozenset()
+                try:
+                    versions_content = tar.extractfile(
+                        f"./{id}/matrix-versions.json",
+                    )
+                    if versions_content:
+                        versions = json.load(versions_content)
+                except KeyError:
+                    STDERR.print(
+                        "No versions detected (couldn't find a "
+                        "'matrix-versions.json' file in ",
+                        f"{id} directory of {input.name}.)",
+                    )
+                    ctx.exit(EX.DATAERR)
+
+                versions_dirs = [
+                    member.name
+                    for member in members
+                    if member.isdir()
+                    and (
+                        member.name.removeprefix(f"./{id}/").removeprefix("v")
```

Comment:
Why are you prepending `v` to the versions -- besides making that a bit awkward that you now have to strip the prefix in a bunch of places, it's also perfectly possible that the version for an implementation *already* starts with `v`, isn't it? What if the implementation claims its version is `v1.2.3` already, and now you added a second `v`?

Reply:
> Why are you prepending v to the versions -- besides making that a bit awkward that you now have to strip the prefix in a bunch of places, it's also perfectly possible that the version for an implementation already starts with v, isn't it? What if the implementation claims its version is v1.2.3 already, and now you added a second v?

Yeah right! Sorry for this. I will have to fix this in the workflows as well since the problem roots from there.

Reply:
So remove the prepended `v` completely from everywhere right ?